### DISCUSSION TITLE
TFT_eSPI newer than v2.2.20 and TTGO T-Display

### DISCUSSION THREAD

**Discussion by uribench** [Bodmer/TFT_eSPI#2655] [Compatible displays and setup files]
I am using Arduino IDE 2.1.0 with ESP32 TTGO T-Display development board with several versions of the TFT_eSPI library with `TTGO T1` board selected.  

Versions 2.2.14 by Bodmer and version 2.2.20 downloaded from `https://github.com/Xinyuan-LilyGO/TTGO-T-Display`, are both working fine. However, newer versions of the TFT_eSPI library do not work well for me. It seems like the screen width is considered wider than the actual 135 pixels.

I am of course pointing to the same board  (setup#25) with `User_Setup_Select.h` for all the versions that I tried (both those that are working and those that do not).

What am I doing wrong?

**Comment by Bodmer** [comment]
I have not seen this problem before. I will try to set up the same hardware and software environment. Run the Colour_Test example sketch here:
https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Test%20and%20diagnostics/Colour_Test/Colour_Test.ino

You should see a complete green rectangle at the edge of the screen. This tests the screen size and offset settings.

Also run the Read_User_Setup sketch and check the serial output is correct.

**Reply by uribench** [reply]
Thank you @Bodmer for your reply. The Colour_Test sketch indeed provides the same results when using TFT_eSPI v2.2.20, as well as v2.2.50. 
However, the attached simple code doesn't. On TFT_eSPI v2.2.20 the text is placed on the screen where expected, but on v2.2.50 it goes partly off screen (see attached two photos).
I have also attached to your request the output of the Read_User_Setup sketch when TFT_eSPI v2.2.50 is used.
Note that I make sure to clear the compilation cache when switching between library versions.

Probably, my code is incomplete and is relying on some implicit defaults that have changed between library versions.

[Read_User_Setup_Output.txt](https://github.com/Bodmer/TFT_eSPI/files/11490059/Read_User_Setup_Output.txt)
[test_v2.2.20_vs_v2.2.50.zip](https://github.com/Bodmer/TFT_eSPI/files/11490063/test_v2.2.20_vs_v2.2.50.zip)
![IMG_6326](https://github.com/Bodmer/TFT_eSPI/assets/32156535/9a9c5afd-3e54-472e-8182-fe00ec372d16)
![IMG_6325](https://github.com/Bodmer/TFT_eSPI/assets/32156535/9c3d926b-ba49-49d9-a032-59a3a0ee651f)

**Reply by uribench** [reply]
I have made some more experiments and if I am not wrong with my conclusions, it seems that setTextDatum() function is not working the same on TFT_eSPI v2.2.20 and on v2.2.50.
In the above code that I provided (test_v2.2.20_vs_v2.2.50) I replaced line 8 from:
`tft.drawString(string, tft.width() / 2, 0);`
to
`tft.drawString(string, tft.width() / 2, tft.fontHeight() / 2);`
and on TFT_eSPI v2.2.20 I got exactly the same results with these two variants. However, on v2.2.50 the results of these two lines are different. 
It seems that v2.2.50 is behaving correctly and more consistent in this regard.
Am I right and this was corrected on the versions following TFT_eSPI v2.2.20?

**Reply by Bodmer** [reply]
The behaviour has changed. Adapt your sketch for v2.2.50

**Reply by uribench** [reply]
Thank you @Bodmer. You have been helpful.
